csrc/dynamic_transform.cpp

```diff
@@ -257,6 +324,65 @@ void DynamicTransformConcretizer::concretizeReshape() {
   }
 }
 
+void DynamicTransformConcretizer::concretizeResize() {
+  // Concretize each resize op.
+  for (const auto& resize_info : info_.getResizeTransforms()) {
+    auto incomplete_out_tv = std::get<0>(resize_info);
+    auto id = std::get<1>(resize_info);
+    auto iter_type = std::get<2>(resize_info);
+
+    auto new_id = IterDomainBuilder(id).iter_type(iter_type).build();
+
+    // swap in new IterDomain as output of the resize Expr
+    ir_utils::replaceValInExpr(id->definition(), id, new_id);
```

Comment:
And maybe what the concretization info should hold would not be the IterType but the extents of the input ID as well as the concrete expansion factors.

Reply:
Holding extents and concrete pad values may be a problem since different combinations can give rise to the same output IterType, which is ultimately the only difference in concretized resizes. We want to have as close to an injective mapping as possible from concretization info to concretized Fusions so that we maximize cache hits.

Reply:
We could use `IterDomain::resize` and pass const left and right pad values. However, I am not sure how to make the input extent const, and we actually don't want it to be since we may have different input sizes for the same concretization (see previous comment).

Reply:
Why not add an optional parameter of `IterType` to `IterDomain::resize` to dictate the output `IterType`? We could just reuse the same expansion factors but this time pass the resolved IterType too. `IterDomain::resize` would just use the given type instead of checking the extents and factors.

Reply:
That's a good idea. It will be cleaner to use `IterDomain::resize` as you said. I'll add the `output_itertype` option and switch to using that.